environment-api/src/test/java/com/sequenceiq/environment/api/v1/environment/model/request/aws/AwsDiskEncryptionParametersTest.java

```diff
@@ -0,0 +1,84 @@
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:012345678910:key/", "arn:aws:kms:us-east-1:012345678910:key/", false},
+
+                {"encryptionKeyArn=aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:abc:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:kms:us-east-1:abc:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws:role:us-east-1:012345678910/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:role:us-east-1:012345678910/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east/2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east/2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-2:0123456789-:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-2:0123456789-:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms::0123456789:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms::0123456789:key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-2::key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-1::key/1234abcd-12ab-34cd-56ef-1234567890ab", false},
+
+                {"encryptionKeyArn=arn:aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+
+                {"encryptionKeyArn=arn:aws-cn:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-cn:kms:us-east-1:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+
+                {"encryptionKeyArn=arn:aws-us-gov:kms:us-east-2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                        "arn:aws-us-gov:kms:us-east-2:012345678910:key/1234abcd-12ab-34cd-56ef-1234567890ab", true},
+        };
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("encryptionKeyArnValidationTestDataProvider")
+    void encryptionKeyArnValidationTest(String testCaseName, String encryptionKeyArn, boolean expectedValid) {
+        if (expectedValid) {
+            assertTrue(AwsDiskEncryptionParameters.AWS_ENCRYPTION_KEY_INVALID_MESSAGE,
```

Comment:
If the assertion fails, then tell so. `AWS_ENCRYPTION_KEY_INVALID_MESSAGE` is for the end-user. Which also means that this constant can be `private static final`
Assertion message should state what and why failed. Like `encryptionKeyArn + " should match pattern " +AwsDiskEncryptionParameters.REG_EXP_FOR_ENCRYPTION_KEY_ARN`, and similar for the negative assert.

Reply:
Addressed